lib/cocoapods-core/lockfile.rb

```diff
@@ -493,7 +493,8 @@ def generate_spec_repos(spec_repos)
         Hash[spec_repos.map do |source, specs|
           next unless source
           next if specs.empty?
-          key = source.url.downcase || source.name.downcase
+          key = source.url || source.name
+          key = key.downcase if source.name == 'master'
```

Comment:
alternatively we could also lowercase all repos that are `if source.git?`. WDYT?

Reply:
In our case we have an internal repo with a git source that we do not want to get lowercased, so I feel it's safer to use "master" (BTW it might be good to use a const for the name).

Reply:
yes would love to get a const for this as I saw plenty of places depending on this string.